spring-integration-xmpp/src/test/java/org/springframework/integration/xmpp/config/XmppConnectionFactoryBeanTests.java

```diff
@@ -46,4 +51,38 @@ public void testXmppConnectionFactoryBeanViaConfig() throws Exception {
 		// the fact that no exception was thrown satisfies this test
 	}
 
+	@Test
+	public void testXmppConnectionFactoryBeanNoRoster() throws Exception {
+		XmppConnectionFactoryBean xmppConnectionFactoryBean = new XmppConnectionFactoryBean() {
+			@Override
+			protected XMPPConnection createInstance() throws Exception {
+				return mock(XMPPTCPConnection.class);
+			}
+		};
+		xmppConnectionFactoryBean.setSubscriptionMode(null);
+		XMPPConnection connection = xmppConnectionFactoryBean.createInstance();
+		assertNotNull(connection);
+		TestUtils.setPropertyValue(xmppConnectionFactoryBean, "connection", connection);
```

Comment:
I'm pretty sure we don't need this functionality.
There is just enough to check the `connection` after `start()`.
The second your test without `setSubscriptionMode(null)` just covers the case when we don't modify `RosterLoadedAtLogin`.
That's all.

I still don't see any documentation changes though...

Reply:
Now I'm confused. I cannot call start like this, the configuration will be null because I extended the factory bean and as the configuration property is private in the super class it will not be set. The test will never run that far without injecting the property first by some means, I will get a NPE. Maybe we are talking about two different things here?

Reply:
Concerning the documentation can you point me to the documentation that describes how you organise it? I could not find it in the github pages. Also I'm not sure what to change as we do not change the documented behaviour. Maybe the only thing I would change is: 

"To switch off the Roster listener for subscriptions you should configure it with..."

to 

"To turn off Roster you should configure it with..."

Reply:
In the root of project there is `/src/reference/asciidoc/xmpp.adoc`. You can find required sentence there. Since we change the behavior a bit (`roster.setRosterLoadedAtLogin(false);`) I really would to mention it in the Doc, even if that was a bug before. Just only like:
```
In addition to the subscription the roster is also disable at login phase. See `Roster.setRosterLoadedAtLogin(Boolean)` for more information.
```

Reply:
Oh! I see what you mean about `connection` inject. The fact that we don't rely there on the `super.getObject()`, but use the local `connection` variable. 

I reworked `XmppConnectionFactoryBean` to use:
```
protected XMPPTCPConnection getConnection() {
		try {
			return (XMPPTCPConnection) getObject();
		}
		catch (Exception e) {
			throw new IllegalStateException("Cannot obtain connection instance", e);
		}
	}
```
instead of local `connection` property and now test-case looks like:
```
public void testXmppConnectionFactoryBeanNoRoster() throws Exception {
		XmppConnectionFactoryBean xmppConnectionFactoryBean = new XmppConnectionFactoryBean() {
			@Override
			protected XMPPConnection createInstance() throws Exception {
				return mock(XMPPTCPConnection.class);
			}
		};

		xmppConnectionFactoryBean.setSubscriptionMode(null);
		xmppConnectionFactoryBean.afterPropertiesSet();
		xmppConnectionFactoryBean.start();
		XMPPConnection connection = xmppConnectionFactoryBean.getObject();

		assertFalse(Roster.getInstanceFor(connection).isRosterLoadedAtLogin());
	}
```
WDYT, please?

Reply:
OK I see where you are heading at. But then we need to replace all references to the configuration property with this new getter, because otherwise it will still be null in the test. But what you risk now is that clients get a bean that has a configuration proxy instead of a properly configured bean. Because now this will work xmppConnectionFactoryBean.setConnectionConfiguration(null).  So I think it is less evil to set the configuration property by reflection.

Reply:
> But what you risk now is that clients get a bean that has a configuration proxy instead of a properly configured bean.

Not sure what you mean here. If you call `afterPropertiesSet()` (and what is definitely done by Spring), you get the properly configured `connection` instance.

Reply:
Maybe I still don't get it, but how I understand you changes we need to do the following as well:
```
	@Override
	public void start() {
           // this.connection.connect(); -> will be null so replace with getter
	    this.getConnection().connect();
```
And getObject() works in a way that you always get a proxy no matter what your bean lifecycle is, because now we are calling it outside the lifecycle anyway. So with these changes in place nothing is preventing a user to insert null instead of a configuration object and the context will start no matter what.

You can test this by adding the following lines to the tests:

```
xmppConnectionFactoryBean.setSubscriptionMode(null);
xmppConnectionFactoryBean.setConnectionConfiguration(null);
xmppConnectionFactoryBean.afterPropertiesSet();
xmppConnectionFactoryBean.setConnectionConfiguration(null);
xmppConnectionFactoryBean.start();
```